### PR TITLE
Metrics API endpoint

### PR DESCRIPTION
**Pull request by ignapas**
# Description

Adds a new endpoint that gathers and returns some required metrics:
* Active users during last one-year period
* Total number of datasets in the portal
* Number of funded projects
* Downloads report

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Manually

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by ignapas** [comment]
New endpoint is in <sparc-api-url>/metrics
Needs a file containing the key to communicate with Google Analytics
Requires new environmental variables to be set (see changes in ``sparc-api/app/config.py``)

**Comment by ignapas** [comment]
Tried to use the installed Pennsieve Python client to get the downloads report but failed to find a way to it... then I saw it requires no authentication so I did a basic GET request

File: app/metrics/algolia.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+APP_ID = Config.ALGOLIA_APP_ID
+API_KEY = Config.ALGOLIA_API_KEY
+
+
+def init_algolia_client():
+    return SearchClient.create(APP_ID, API_KEY)
+
+
+def get_dataset_count(client):
+    index = client.init_index("k-core_dev")
```
Comment (egauzens) [inline app/metrics/algolia.py:13]: "k-core_dev" should be an environment variable. For staging we can look at k-core_dev, but for prod we should be looking at SPARC_pr

**Review by egauzens — CHANGES_REQUESTED** [review]
looks good, except for the comment I made

File: app/metrics/algolia.py (hunk 1 above)
Comment (ignapas) [reply app/metrics/algolia.py]: oups sorry! fixed now

**Comment by ignapas** [comment]
@alan-wu could you please take a look at the GA part? thank you

**Review by egauzens — APPROVED** [review]

**Review by alan-wu — APPROVED** [review]
@ignapas Thanks, looks good to me. Can you please provide a link to the wrike ticket in the future when possible? It will help the reviewers to see the bigger picture.

**Review by egauzens — APPROVED** [review]

**Pull request merged**